TextAnalyserProject/src/Controllers/FileController.java

```diff
@@ -57,4 +56,104 @@ public void printLettersInFileCount() {
+
+
+        try (BufferedInputStream in = new BufferedInputStream(new URL("https://s3.zylowski.net/public/input/8.txt").openStream())) {
+            try (FileOutputStream fileOutputStream = new FileOutputStream("plik.txt")) {
+                byte dataBuffer[] = new byte[1024];
+                int bytesRead;
+
+                while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
+                    fileOutputStream.write(dataBuffer, 0, bytesRead);
+                }
+            }
+        } catch (IOException e) {
+            // handle exception
+        }
+
+    }
+
+
+    public static int punctuationMarkCount() {
+        String file = "plik.txt";
+        String words = null;
+        try {
+            words = new String(Files.readAllBytes(Paths.get(file)));
+        } catch (IOException e) {
+            throw new NullPointerException("File not found");
+        }
+        int number = 0;
+        char array[] = {',','.', ';', ':','-', '?', '!', '"', '(', ')'};
+        for (int i = 0, length = words.length(); i < length; i++) {
+            if (new String(array).contains(String.valueOf(words.charAt(i)))) {
+
+                number++;
+            }
+
+        }
+        return number;
+    }
+  
+    public static int charCount() {
+        String file = "plik.txt";
+        String words = null;
+        try {
+            words = new String(Files.readAllBytes(Paths.get(file)));
+        } catch (IOException e) {
+            throw new NullPointerException("File not found");
+        }
+        int number = 0;
+        for (int i = 0, length = words.length(); i < length; i++) {
+            if (words.charAt(i) != ' ') {
+                number++;
+            }
+        }
+        return number;
+
+    }
+
+    public static int wordCount()
+
+      {
+            String file = "plik.txt";
+            String words = null;
+            try {
+                words = new String(Files.readAllBytes(Paths.get(file)));
+            } catch (IOException e) {
+                throw new NullPointerException("File not found");
+            }
+            String[] w = words.split("\\s");
+            return w.length - 1;
+        }
+
+
+//    public static int letterCount() {
+//        String file = "plik.txt";
+//        String words = null;
+//        try {
+//            words = new String(Files.readAllBytes(Paths.get(file)));
+//        } catch (IOException e) {
+//            throw new NullPointerException("File not found");
+//        }
+//        int number = 0;
+//        String letter = "abcdefghijk";
+//        for (int i = 0, length = words.length(); i < length; i++) {
+//            if (new String(letter).contains(String.valueOf(words.charAt(i)))) {
+//
+//                number++;
+//            }
+//
+//        }
+//        return number;
+
+    public static void deleteFile(String filePath) {
+        File file = new File(filePath);
+        if (!file.exists()) {
+          System.out.println("Plik nie istnieje");
```

Comment:
Każde wyłączenie programu, podczas którego nie próbowano pobrać pliku, wyświetli komunikat "Plik nie istnieje". Czy to zamierzone działanie?